### DISCUSSION TITLE
I want to create an edit-action to modify some selected text. How can I do this?

### DISCUSSION THREAD

**Discussion by pouyakary** [zed-industries/zed#9769] [Feature Requests]
Hi amazing editor's community.

I have an extension on vscode, where I can justify comments to be more readable, and I want to implement it inside Zed as well:

<img width="729" alt="Screenshot 1403-01-06 at 11 52 52 AM" src="https://github.com/zed-industries/zed/assets/2157285/3a6ff7b3-1b3a-471b-a270-4023f95d108e">

I discovered the `zed_extension_api` crate in the source code, but the documentation was little. What I need is a few things:

- a way to register a command
- to get the current buffer
- know what line the cursor is at on
- be able to read lines by their line number
- be able to change a selection of lines (say from 9 to 11)
- detect either the language id or if possible, know the single line comment style of the current language, say "#" for python, "//" for C.

And another question is if the extensions aren't ready yet, where should I add it in the source code?

**Comment by matj1** [comment]
I too would like to create a command which edits text around the cursor, but I can't find functions in the api which allow it. The command is a port of just-one-space in Emacs.

**Reply by failable** [reply]
Not currently possible. You can use task for that, but you will have to copy the output back to the buffer.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
